lib/paypal_express/api.rb

```diff
@@ -29,20 +29,72 @@ def on_event(event)
       end
 
       def authorize_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_payment_method_id, amount, currency, properties, context)
```

Comment:
Would it be possible to share the code (it's pretty much the same as `purchase_payment`)?

AFAICT, the main differences are just:

* `transaction_type` and `api_call` (in non HPP flow) when creating the response row
* The `ActiveMerchant` call in the gateway proc